internal/http/services/appprovider/appprovider.go

```diff
@@ -135,34 +126,87 @@ func (s *svc) handleNew(w http.ResponseWriter, r *http.Request) {
 
 	client, err := pool.GetGatewayServiceClient(s.conf.GatewaySvc)
 	if err != nil {
-		ocmd.WriteError(w, r, ocmd.APIErrorServerError, "error getting grpc gateway client", err)
+		writeError(w, r, appErrorServerError, "error getting grpc gateway client", err)
 		return
 	}
 
 	if r.URL.Query().Get("template") != "" {
 		// TODO in the future we want to create a file out of the given template
-		ocmd.WriteError(w, r, ocmd.APIErrorInvalidParameter, "Template not implemented",
-			errtypes.NotSupported("Templates are not yet supported"))
+		writeError(w, r, appErrorUnimplemented, "template is not implemented", nil)
+		return
+	}
+
+	parentContainerID := r.URL.Query().Get("parent_container_id")
+	if parentContainerID == "" {
+		writeError(w, r, appErrorInvalidParameter, "missing parent container ID", nil)
+		return
+	}
+
+	parentContainerRef := resourceid.OwnCloudResourceIDUnwrap(parentContainerID)
+	if parentContainerRef == nil {
+		writeError(w, r, appErrorInvalidParameter, "invalid parent container ID", nil)
+		return
+	}
+
+	filename := r.URL.Query().Get("filename")
+	if filename == "" {
+		writeError(w, r, appErrorInvalidParameter, "missing filename", nil)
+		return
+	}
+
+	dirPart, filePart := path.Split(filename)
+	if dirPart != "" || filePart != filename {
```

Comment:
Not sure about this. We support requests like `storageID:nodeID!filename` but not `storageID:nodeID!some-dir/filename`. If we say that relative paths are supported, both should be allowed in my opinion

Reply:
The `/app/new` requests requires two query parameters:
- `parent_container_id`
- `filename`

The code ensures that `filename` is only a filename and does not contain a path segment. I don't see any reason why we should support relative paths in that particular case since we would then need to handle non existent path segments and much more corner cases.  The container referenced by `parent_container_id` can be easily checked if it exists and permissions are sufficient.

I don't see any real world uses cases for these examples:
Example 1:
- `parent_container_id` = `MTI4NGQ...==` => reference to `/home/test/documents`
- `filename` = `../../Document1.odt`

-> one could easily take the reference of `/home` in the first place

Example 2:
- `parent_container_id` = `MTI4NGQ...==` => reference to `/home/test/documents`
- `filename` = `./office/Document2.odt`
-> we would need to check that the folder `/home/test/documents/office` actually exists, which adds extra complexity

Reply:
Sure, the web client doesn't need to use this format but I meant for the sake of consistency since these types of requests are supported elsewhere. Adding the extra check sounds okay to me if it means we're consistent across services

Reply:
I'll merge the PR now and create an issue to keep a track of this part